interfaces/cython/cantera/composite.py

```diff
@@ -523,9 +523,9 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
                 if not np.shape(v):
-                    self._extra[name] = [v]*self._shape[0]
+                    self._extra[name] = np.array([v]*self._shape[0])
```

Comment:
I know this is a pre-existing choice, but It may make sense to allow for more dimensions? I.e. this would be a case where we have something like
```
   >>> states = ct.SolutionArray(gas, (6, 10), extra=[‘foo’, ‘bar’])
```

Reply:
I can't clearly understand what you want to say. I guess you want foo and bar be extra arrays of size (6,10)?
like 
```
>>> states = ct.SolutionArray(gas, (6,10), extra={'foo':sd,"bar":sd})
```
where sd is any (6,10) list/np array
??

Reply:
@sin-ha ... what I wrote was deliberate - no typo there. It currently throws a `ValueError`, but you could argue that it should initialize a 'np.ndarray' with the full `_shape`, not just the first dimension.

Reply:
@ischoegl I'm also not sure what you're suggesting, but it seems like it would be a new feature? In that case, I think it seems outside the scope of this fix.

Reply:
As of now extra expects dicts unless the shape of SolutionArray is trivial
- If the user passes a list we should be able to initiate the extra array with same shape.
- else the user will pass that as a dict , in that case we should validate extra with the complete SolutionArray shape instead of ```self._shape[0]```

Reply:
I also do think this is more of adding another functionality, seems a bit off  of the issue

Reply:
I think the point I raised is a trivial addition, and the application was stated in my first post: it’s a reasonable assumption that the arrays are instantiated based on the information provided. My guess is that the former single-dimensioned approach was based on `_extra` formerly being list-based. Now that this is no longer the case, putting in a more consistent approach is trivial.

What I am talking about affects some of the same lines that are changed already and won’t take more than a couple of additional lines. Hence, I’d like to see it considered as a friendly amendment of the original issue 😉